src/components/application_manager/test/commands/hmi/hmi_notifications/hmi_notifications_test.cc

```diff
@@ -0,0 +1,2125 @@
+  for (; it != arg1.group_permissions.end(); ++it) {
+    consent->group_permissions.push_back(*it);
+  }
+  consent->consent_source = arg1.consent_source;
+}
+ACTION_P2(GetBoolValueAndString, value, str) {
+  *value = arg0;
+  *str = arg1;
+}
+ACTION_P(GetMessage, message) {
+  (*message)[am::strings::params][am::strings::function_id] =
+      (*arg0)[am::strings::params][am::strings::function_id].asInt();
+  (*message)[am::strings::params][am::strings::message_type] =
+      (*arg0)[am::strings::params][am::strings::message_type].asInt();
+  (*message)[am::strings::params][am::strings::correlation_id] =
+      (*arg0)[am::strings::params][am::strings::correlation_id].asInt();
+  (*message)[am::strings::params][am::strings::connection_key] =
+      (*arg0)[am::strings::params][am::strings::connection_key].asInt();
+  return (*message)[am::strings::params][am::strings::correlation_id].asInt();
+}
+
+class HMICommandsNotificationsTest
+    : public components::commands_test::CommandsTest<
+          CommandsTestMocks::kIsNice> {
+ public:
+  HMICommandsNotificationsTest()
+      : applications_(application_set_, applications_lock_) {}
+
+ protected:
+  am::ApplicationSet application_set_;
+  sync_primitives::Lock applications_lock_;
+  DataAccessor<am::ApplicationSet> applications_;
+  MockHMICapabilities mock_hmi_capabilities_;
+
+  NiceMock<event_engine_test::MockEventDispatcher> mock_event_dispatcher_;
+  NiceMock<policy_test::MockPolicyHandlerInterface> policy_interface_;
+
+  std::vector<utils::SharedPtr<am::commands::Command> >
+      commands_list_manage_mobile_;
+  std::vector<MessageSharedPtr> messages_list_for_commands_manage_mobile_;
+
+  std::vector<utils::SharedPtr<am::commands::Command> > commands_list_to_hmi_;
+  std::vector<MessageSharedPtr> messages_list_for_commands_to_hmi_;
+
+  std::vector<hmi_apis::FunctionID::eType>
+      id_list_for_commands_event_dispatcher_;
+  std::vector<utils::SharedPtr<am::commands::Command> >
+      commands_list_event_dispatcher_;
+
+  application_manager_test::MockStateController mock_state_controller_;
+
+  const uint32_t kCountCommandsManageMobile_ = 22;
+  const uint32_t kCountCommandsToHmi_ = 12;
+
+  const int32_t kHmiProtocolType_ = 1;
+  const int32_t kMobileProtocolType_ = 0;
+  const int32_t kProtocolVersion_ = 3;
+
+  const uint32_t kCorrelationId_ = 1939;
+
+  am::ApplicationSharedPtr app_;
+  NiceMock<MockApplication>* app_ptr_;
+
+  const uint32_t kAppId_ = 2014;
+
+  void FillMessagesListCommandsForManageMobile() {
+    for (uint32_t i = 0; i < kCountCommandsManageMobile_; ++i) {
+      messages_list_for_commands_manage_mobile_.push_back(CreateMessage());
+    }
+  }
+
+  void FillCommandsListForManageMobile() {
+    std::vector<MessageSharedPtr>::iterator it =
+        messages_list_for_commands_manage_mobile_.begin();
+
+    commands_list_manage_mobile_.push_back(
+        CreateCommand<am::commands::OnVIAccPedalPositionNotification>((*it)));
+    ++it;
```

Comment:
@okozlovlux These three lines are repetitive with only one argument changing. Perhaps a macro or template will be useful here.